lib/Chunk.js

```diff
@@ -133,9 +138,72 @@ class Chunk {
 		});
 	}
 
+	setModules(modules) {
+		this._modules = new Set(modules);
+		this._modulesIsSorted = false;
+	}
+
+	getNumberOfModules() {
+		return this._modules.size;
+	}
+
+	get modulesIterable() {
+		return this._modules;
+	}
+
+	forEachModule(fn) {
+		this._modules.forEach(fn);
+	}
+
+	mapModules(fn) {
+		const modules = this._modules;
+		const array = new Array(modules.size);
+		let idx = 0;
+		for(let module of modules) {
+			array[idx++] = fn(module, idx, modules);
+		}
+		return array;
```

Comment:
Array.from has a mapping function argument:

```js
const modules = this._modules
return Array.from(modules, (module, idx) => fn(module, idx, modules))
```

By the way, should the callback get the Set itself as the 3rd argument?